pandas/core/groupby/generic.py

```diff
@@ -1100,13 +1100,19 @@ def py_fallback(bvalues: ArrayLike) -> ArrayLike:
             #  in the operation.  We un-split here.
             result = result._consolidate()
             assert isinstance(result, (Series, DataFrame))  # for mypy
-            mgr = result._mgr
-            assert isinstance(mgr, BlockManager)
-            assert len(mgr.blocks) == 1
 
             # unwrap DataFrame to get array
-            result = mgr.blocks[0].values
-            return result
+            mgr = result._mgr
+            assert isinstance(mgr, BlockManager)
+            if len(mgr.blocks) != 1:
+                # We've split an object block! Everything we've assumed
+                # about a single block input returning a single block output
+                # is a lie. See eg GH-39329
+                return mgr.as_array()
+            else:
+                assert len(mgr.blocks) == 1
```

Comment:
is the assert now redundant (or needed for mypy)?

Reply:
i think unnecessary

Reply:
Removed the assert